src/Microsoft.Health.Fhir.SqlServer/Features/Exceptions/RecordNotFoundException.cs

```diff
@@ -0,0 +1,26 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using EnsureThat;
+
+using Microsoft.Health.Fhir.Core.Exceptions;
+using Microsoft.Health.Fhir.Core.Models;
+
+namespace Microsoft.Health.Fhir.SqlServer.Features.Exceptions
+{
+    public class RecordNotFoundException : FhirException
```

Comment:
This probably shouldn't extend `FhirException` since it doesn't actually deal with FHIR.